src/appengine/handlers/cron/fuzz_strategy_selection.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Fuzzing strategy selection cron job.
+
+Runs multi-armed bandit experiments for fuzzing strategy selection.
+In particular, this is a Boltzman Exploration (softmax) implementaion
+of multi-armed bandit experiments. Queries from bigquery to update
+multi-armed bandit probability values based on the new edges for various
+combined strategies. In the upload_bandit_weights function, we can change
+metric to be for edges, crash, features, or units. Currently based on new
+edges."""
+
+from datastore import data_types
+from datastore import ndb
+from google_cloud_utils import big_query
+from handlers import base_handler
+from libs import handler
+
+# BigQuery query for calculating multi-armed bandit probabilities for
+# various strategies using a Boltzman Exploration (softmax) model.
+
+# Averages standardized new_edges feature over each strategy for expected
+# new_edges metric for each strategy.
+# See https://www.cs.mcgill.ca/~vkules/bandits.pdf for formula.
+
+MULTI_ARMED_BANDIT_PROB = """
```

Comment:
I think I might have closed out the comment where I mentioned this unintentionally, but this query name still seems a bit odd to me. Without the rest of the context it doesn't really seem like a constant query, but more like a fixed probability. Could this just be BANDIT_PROBABILITY_QUERY or something along those lones?